x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java

```diff
@@ -167,7 +168,13 @@ ResolvedIndices resolveIndicesAndAliases(String action, IndicesRequest indicesRe
                     indicesReplacedWithNoIndices = true;
                     resolvedIndicesBuilder.addLocal(NO_INDEX_PLACEHOLDER);
                 } else {
-                    throw new IndexNotFoundException(Arrays.toString(indicesRequest.indices()));
+                    final String[] indices = indicesRequest.indices();
+                    if (indices != null && indices.length == 1 && Regex.isMatchAllPattern(indices[0])) {
+                        // Special handling to keep error message the same as previously thrown in core's IndexAbstractionResolver
+                        throw new IndexNotFoundException(indices[0]);
```

Comment:
This logic here just to throw index not found for `*` instead of index not found `[*]`?
I wouldn't bother...

Reply:
I wouldn't bother either, but we have tests for it:
* https://github.com/elastic/elasticsearch/blob/324996ed3265d3046f4e63230e666f31275212bc/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authz/ReadActionsTests.java#L105
* https://github.com/elastic/elasticsearch/blob/324996ed3265d3046f4e63230e666f31275212bc/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authz/ReadActionsTests.java#L133

Also EQL has raised a [few](#63561) [issues](#63295) in the past that are mostly related to the different behaviours between core and security enabled. I was concerned that changing the error message (from `no such index [*]` to `no such index [[*]]`), even though minor, could be "breaking" in some sense.